applications/AdjointFluidApplication/custom_utilities/numerical_diffusion.h

```diff
@@ -0,0 +1,438 @@
+//  KratosAdjointFluidApplication
+//
+//  License:		 BSD License
+//					 license: AdjointFluidApplication/license.txt
+//
+//  Main authors:    Suneth Warnakulasriya, https://github.com/sunethwarna
+//
+
+#if !defined(KRATOS_NUMERICAL_DIFFUSION)
+#define KRATOS_NUMERICAL_DIFFUSION
+
+// System includes
+#include <vector>
+#include <string>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/element.h"
+#include "includes/process_info.h"
+#include "includes/model_part.h"
+#include "includes/ublas_interface.h"
+#include "includes/kratos_parameters.h"
+#include "utilities/openmp_utils.h"
+
+// Eigen library includes
+#include <Eigen/SVD>
```

Comment:
if you include this without checking if it is available then the compilation of this file should be optional => flag or so in CMakeLists